caikit_nlp/modules/text_generation/text_generation_local.py

```diff
@@ -0,0 +1,247 @@
+    name="Text Generation",
+    version="0.1.0",
+    task=TextGenerationTask,
+)
+class TextGeneration(ModuleBase):
+    """Module to provide text generation capabilities"""
+
+    supported_resources = [HFAutoCausalLM, HFAutoSeq2SeqLM]
+
+    def __init__(
+        self,
+        base_model_name: str,
+        base_model: PretrainedModelBase = None,
+        eos_token: str = None,
+    ):
+        super().__init__()
+
+        error.type_check("<NLP48137045E>", str, allow_none=True, eos_token=eos_token)
+        self.base_model = base_model
+        self.base_model_name = base_model_name
+
+        # Set _model_loaded as False by default. This will only get set to True if
+        # we enable the tgis_backend and we are able to fetch the client successfully.
+        self._model_loaded = False
+        self._eos_token = eos_token
+
+    # pylint: disable=duplicate-code
+    def __del__(self):
+        del self.base_model
+        gc.collect()
+        try:
+            torch.cuda.empty_cache()
+        except AttributeError:
+            pass
+
+    @classmethod
+    def bootstrap(cls, base_model_path: str):
+        """Function to bootstrap a pre-trained transformers model and
+        get a caikit text-generation 'model'.
+
+        Args:
+            base_model_path: str
+                Path to transformers model
+                NOTE: Model path needs to contain tokenizer as well
+        Returns:
+            caikit_nlp.blocks.text_generation.TextGeneration
+                Object of TextGeneration class (model)
+        """
+        # pylint: disable=duplicate-code
+        model_config = AutoConfig.from_pretrained(base_model_path)
+
+        resource_type = None
+        for resource in cls.supported_resources:
+            if model_config.model_type in resource.SUPPORTED_MODEL_TYPES:
+                resource_type = resource
+                break
+
+        if not resource_type:
+            error(
+                "<NLP61784225E>",
+                "{} model type is not supported currently!".format(
+                    model_config.model_type
+                ),
+            )
+        log.debug("Bootstrapping base resource [%s]", base_model_path)
+        base_model = resource_type.bootstrap(
+            base_model_path, tokenizer_name=base_model_path
+        )
+        eos_token = base_model._tokenizer.eos_token or None
+        return cls(
+            base_model_path,
+            base_model,
+            eos_token=eos_token,
+        )
+
+    @classmethod
+    def load(cls, model_path: str) -> "TextGeneration":
+        """Function to load text-generation model
+
+        Args:
+            model_path: str
+                Path to the model to be loaded.
+        Returns:
+            TextGeneration
+                Instance of this class built from the on disk model.
+        """
+
+        config = ModuleConfig.load(model_path)
+        base_model_path = config.get("artifact_path", "")
```

Comment:
Why is there a fallback here? Do we expect a caikit config file to be added directly to HF models?

Reply:
good point. There is a potential use-case of this in the realms of using HF models directly, but for the moment, that is not target, so it would be good to explicitly fail, instead of magically work, I'll remove this.

@Ssukriti please let me know if you had other thoughts?